udata/api/oauth2.py

```diff
@@ -161,6 +165,13 @@ def get_expires_at(self):
     def get_client_id(self):
         return str(self.client.id)
 
+    def is_expired(self):
```

Comment:
I'm slightly confused between `is_expired` VS `is_refresh_token_valid` logic

Reply:
`is_expired` seems to be a new method that needs to be implemented in 1.3.1, and is related to the token: is the (access) token still good, or is it expired?

The **refresh** token itself needs to be checked to see if it's still valid. It's valid as long as it's not expired (it's longer lived than the access token), AND has not been revoked.

This allows the revoking mecanism to work properly. Without a short lived access token, even if it's revoked on the auth server (eg AgentConnect's server), the client would still have access to the service (eg datagouv) as long as the access token is alive.

Making the access token short lived forces the client to query for a new one every so often, or to refresh it, at which point the revocation is checked.

I'm not sure if I'm being clear here, but I would be glad to explain this in more details if needed.

Reply:
Yes I think I get the idea now, thank you! :pray: